src/sync/app.hpp

```diff
@@ -269,6 +269,23 @@ class App {
      * Log out the given user if they are not already logged out.
      */
     void log_out(std::shared_ptr<SyncUser> user, std::function<void(Optional<AppError>)> completion_block) const;
+    
+    /// Switches the active user to the user with the specified id. The user must
+    /// exist in the list of all users who have logged into this application, and
+    /// the user must be currently logged in, otherwise this will return an
+    /// AppError.
+    ///
+    /// @param user_id A String specifying the desired `user_id`
+    /// @param completion_block will return a SyncUser on success, an AppError on failure
+    void switch_user(const std::string& user_id,
+                     std::function<void(std::shared_ptr<SyncUser>, Optional<AppError>)> completion_block) const;
+    
+    /// Logs out and removes the user with the provided id
+    /// if no user ID is provided it will attempt to log out the current user and remove it
```

Comment:
It would probably be cleaner to have a public `remove_user` method that only accepts a completion block similar to how log_out works. This feels potentially confusing.

Reply:
I've now added another method that just takes the completion block